packages/rocketchat-file-upload/server/lib/responseS3.js

```diff
@@ -0,0 +1,28 @@
+var crypto = Npm.require("crypto");
+
+var S3bucket, S3accessKey, S3secretKey;
+
+RocketChat.settings.get('FileUpload_S3_Bucket', function(key, value) {
+	S3bucket = value;
+});
+RocketChat.settings.get('FileUpload_S3_AWSAccessKeyId', function(key, value) {
+	S3accessKey = value;
+});
+RocketChat.settings.get('FileUpload_S3_AWSSecretAccessKey', function(key, value) {
+	S3secretKey = value;
+});
+
+var generateURL = function(fullUrl) {
+	var resourceURL = '/' + fullUrl.substr(fullUrl.indexOf(S3bucket));
```

Comment:
resourceURL gives string like that for me `/bucket-name.s3.amazonaws.com/ZkchB8J9Nk3neCEHy/HcY5AuFM66M5hkfX6/esH5Y2FiiWvav8yBW`
which gives wrong signature and 403 error.
resourceURL should be like that `/bucket-name/ZkchB8J9Nk3neCEHy/HcY5AuFM66M5hkfX6/esH5Y2FiiWvav8yBW`

Possible fix:
`var resourceURL = '/' + S3bucket + fullUrl.substr(fullUrl.indexOf('s3.amazonaws.com')+16);`

@sampaiodiego please have a look.